lib/internal/js_stream_socket.js

```diff
@@ -141,7 +141,7 @@ class JSStreamSocket extends Socket {
 
     const handle = this._handle;
 
-    setImmediate(() => {
+    process.nextTick(() => {
```

Comment:
Why this change? Do this update break something related to `setImmediate` or is it just an unrelated change?

Reply:
https://github.com/nodejs/node/pull/42340#issuecomment-1261163284

Reply:
Should we land this as a separate commit to make sure it's not breaking anything? I mean I'm OK with keeping it as one commit in this PR if you don't think it's necessary, but that would seem to be less risky to land it separately.

Reply:
It is already a dedicated commit.

Reply:
The commits should probably be re-ordered, so the fix lands before the libuv upgrade? Anyway, I don't feel strongly either, feel free to land it as you please.

Reply:
https://github.com/nodejs/node/blob/3296d6d23d90151664fe1e3884a198dfb47cb8eb/doc/contributing/pull-requests.md#L540-L541

We can either squash the commits, or change the order, but we shouldn't land as is.

Reply:
Reordered adding some refs to commit messages for future readers.